overlord/snapstate/autorefresh.go

```diff
@@ -407,6 +413,38 @@ func canContinueAutoRefresh(st *state.State) (int, bool) {
 	return 0, false
 }
 
+func (m *autoRefresh) restoreMonitoring() error {
+	if m.restoredMonitoring {
+		return nil
+	}
+
+	var monitored []string
+	if err := m.state.Get("monitored-snaps", &monitored); err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+
+	defer func() { m.restoredMonitoring = true }()
+	if len(monitored) == 0 {
+		return nil
+	}
+
+	monitoring := make(map[string]chan<- bool)
+	for _, snap := range monitored {
+		done := make(chan string, 1)
+		if err := cgroupMonitorSnapEnded(snap, done); err != nil {
```

Comment:
What will happen here if a previously monitored app got closed during the snapd restart? Will that result in an error or is cgroupMonitorSnapEnded just ignoring empty cgroups in this case?

Reply:
It should just immediately send something through the channel and work transparently for the reader